enterprise/internal/batches/store/batch_changes.go

```diff
@@ -204,13 +204,21 @@ func countBatchChangesQuery(opts *CountBatchChangesOpts, repoAuthzConds *sqlf.Qu
 		preds = append(preds, sqlf.Sprintf("changesets.id = %s", opts.ChangesetID))
 	}
 
-	switch opts.State {
-	case btypes.BatchChangeStateOpen:
-		preds = append(preds, sqlf.Sprintf("batch_changes.closed_at IS NULL AND batch_changes.last_applied_at IS NOT NULL"))
-	case btypes.BatchChangeStateClosed:
-		preds = append(preds, sqlf.Sprintf("batch_changes.closed_at IS NOT NULL"))
-	case btypes.BatchChangeStateDraft:
-		preds = append(preds, sqlf.Sprintf("batch_changes.last_applied_at IS NULL"))
+	if len(opts.States) > 0 {
+		stateConds := []*sqlf.Query{}
+		for _, state := range opts.States {
+			switch state {
+			case btypes.BatchChangeStateOpen:
+				stateConds = append(stateConds, sqlf.Sprintf("batch_changes.closed_at IS NULL AND batch_changes.last_applied_at IS NOT NULL"))
+			case btypes.BatchChangeStateClosed:
+				stateConds = append(stateConds, sqlf.Sprintf("batch_changes.closed_at IS NOT NULL"))
+			case btypes.BatchChangeStateDraft:
+				stateConds = append(stateConds, sqlf.Sprintf("batch_changes.last_applied_at IS NULL"))
+			}
+		}
+		if len(stateConds) > 0 {
+			preds = append(preds, sqlf.Sprintf("(%s)", sqlf.Join(stateConds, " OR ")))
```

Comment:
super nitpicky: Pretty sure the spacing is handled by sqlf

```suggestion
			preds = append(preds, sqlf.Sprintf("(%s)", sqlf.Join(stateConds, "OR")))
```